interface/main/main_screen.php

```diff
@@ -15,14 +15,169 @@
  *
  * @package OpenEMR
  * @author  Brady Miller <brady.g.miller@gmail.com>
+ * @author  Rod Roark <rod@sunsetsystems.com>
  * @link    http://www.open-emr.org
  */
 
+/* Include our required headers */
+require_once('../globals.php');
 
+use OpenEMR\Core\Header;
 
+///////////////////////////////////////////////////////////////////////
+// Functions to support MFA.
+///////////////////////////////////////////////////////////////////////
 
-/* Include our required headers */
-require_once('../globals.php');
+function posted_to_hidden($name)
+{
+    if (isset($_POST[$name])) {
+        echo "<input type='hidden' name='" . attr($name) . "' value='" . attr($_POST[$name]) . "' />\n";
+    }
+}
+
+function generate_html_start($title)
+{
+    global $appId, $css_header;
+    ?>
+<html>
+<head>
+    <?php Header::setupHeader(); ?>
+<title><?php echo text($title); ?></title>
+<script src="<?php echo $GLOBALS['webroot'] ?>/library/js/u2f-api.js"></script>
```

Comment:
We can add this asset via npm (https://github.com/openemr/openemr/blob/master/package.json#L45-L90). Just let me know, and I'd be glad to add this.

Reply:
Sounds promising but I'll need to see what's in it. There are some different versions of u2f-api.js floating around and some don't work.

Reply:
I'll send you to the npm u2f-api.js package site when npm website is back up (btw, it can't be a good thing that the npm website is now down...)
![image](https://user-images.githubusercontent.com/278968/45787481-77d69500-bc2a-11e8-9dc5-04788c2f0377.png)

Reply:
ok, here it is:
https://www.npmjs.com/package/u2f-api
(agree there are several; where did you get the version you are using?)